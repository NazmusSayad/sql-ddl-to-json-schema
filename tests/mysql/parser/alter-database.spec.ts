import { getParsedFormat } from '../parse-handler';
import { run } from '../runner';

run(getParsedFormat, {
  'Parser: Should alter test database with charset utf8': {
    queries: [
      'ALTER DATABASE test DEFAULT CHARACTER SET utf8;',
      'ALTER SCHEMA test CHARACTER SET utf8;',
      'ALTER database test CHARACTER SET = "utf8";',
      'ALTER schema test CHARACTER SET =`utf8`;',
      "ALTER DATABASE test CHARACTER SET= 'utf8';",
    ],
  },

  'Parser: Should alter test database with collation utf8_general_ci': {
    queries: [
      'ALTER DATABASE test COLLATE utf8_general_ci;',
      'ALTER DATABASE test DEFAULT COLLATE = `utf8_general_ci`;',
      'ALTER DATABASE test DEFAULT COLLATE ="utf8_general_ci";',
      "ALTER DATABASE test DEFAULT COLLATE= 'utf8_general_ci';",
    ],
  },

  'Parser: Should alter test database with charset utf8 and collation utf8_general_ci': {
    queries: ['alter DATABASE test CHARACTER SET utf8 COLLATE utf8_general_ci;'],
  },

  'Parser: Should create test database with collation utf8_general_ci and charset utf8': {
    queries: ['alter DATABASE test COLLATE utf8_general_ci CHARACTER SET utf8;'],
  },

  'Parser: Should create test database even when having two collate options.': {
    queries: [
      'alter DATABASE test COLLATE utf8_cirylic_ci CHARACTER SET utf8 COLLATE utf8_general_ci;',
    ],
  },
});
