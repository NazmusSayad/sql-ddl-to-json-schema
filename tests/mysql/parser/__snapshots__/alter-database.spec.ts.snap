// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Parser: Should alter test database with charset utf8 1`] = `
{
  "id": "MAIN",
  "def": [
    {
      "id": "P_DDS",
      "def": {
        "id": "P_ALTER_DB",
        "def": {
          "database": "test",
          "meta": [
            {
              "id": "O_ALTER_DB_SPEC",
              "def": {
                "charset": "utf8"
              }
            }
          ]
        }
      }
    }
  ]
}
`;

exports[`Parser: Should alter test database with charset utf8 2`] = `
{
  "id": "MAIN",
  "def": [
    {
      "id": "P_DDS",
      "def": {
        "id": "P_ALTER_DB",
        "def": {
          "database": "test",
          "meta": [
            {
              "id": "O_ALTER_DB_SPEC",
              "def": {
                "charset": "utf8"
              }
            }
          ]
        }
      }
    }
  ]
}
`;

exports[`Parser: Should alter test database with charset utf8 3`] = `
{
  "id": "MAIN",
  "def": [
    {
      "id": "P_DDS",
      "def": {
        "id": "P_ALTER_DB",
        "def": {
          "database": "test",
          "meta": [
            {
              "id": "O_ALTER_DB_SPEC",
              "def": {
                "charset": "utf8"
              }
            }
          ]
        }
      }
    }
  ]
}
`;

exports[`Parser: Should alter test database with charset utf8 4`] = `
{
  "id": "MAIN",
  "def": [
    {
      "id": "P_DDS",
      "def": {
        "id": "P_ALTER_DB",
        "def": {
          "database": "test",
          "meta": [
            {
              "id": "O_ALTER_DB_SPEC",
              "def": {
                "charset": "utf8"
              }
            }
          ]
        }
      }
    }
  ]
}
`;

exports[`Parser: Should alter test database with charset utf8 5`] = `
{
  "id": "MAIN",
  "def": [
    {
      "id": "P_DDS",
      "def": {
        "id": "P_ALTER_DB",
        "def": {
          "database": "test",
          "meta": [
            {
              "id": "O_ALTER_DB_SPEC",
              "def": {
                "charset": "utf8"
              }
            }
          ]
        }
      }
    }
  ]
}
`;

exports[`Parser: Should alter test database with charset utf8 and collation utf8_general_ci 1`] = `
{
  "id": "MAIN",
  "def": [
    {
      "id": "P_DDS",
      "def": {
        "id": "P_ALTER_DB",
        "def": {
          "database": "test",
          "meta": [
            {
              "id": "O_ALTER_DB_SPEC",
              "def": {
                "charset": "utf8"
              }
            },
            {
              "id": "O_ALTER_DB_SPEC",
              "def": {
                "collation": "utf8_general_ci"
              }
            }
          ]
        }
      }
    }
  ]
}
`;

exports[`Parser: Should alter test database with collation utf8_general_ci 1`] = `
{
  "id": "MAIN",
  "def": [
    {
      "id": "P_DDS",
      "def": {
        "id": "P_ALTER_DB",
        "def": {
          "database": "test",
          "meta": [
            {
              "id": "O_ALTER_DB_SPEC",
              "def": {
                "collation": "utf8_general_ci"
              }
            }
          ]
        }
      }
    }
  ]
}
`;

exports[`Parser: Should alter test database with collation utf8_general_ci 2`] = `
{
  "id": "MAIN",
  "def": [
    {
      "id": "P_DDS",
      "def": {
        "id": "P_ALTER_DB",
        "def": {
          "database": "test",
          "meta": [
            {
              "id": "O_ALTER_DB_SPEC",
              "def": {
                "collation": "utf8_general_ci"
              }
            }
          ]
        }
      }
    }
  ]
}
`;

exports[`Parser: Should alter test database with collation utf8_general_ci 3`] = `
{
  "id": "MAIN",
  "def": [
    {
      "id": "P_DDS",
      "def": {
        "id": "P_ALTER_DB",
        "def": {
          "database": "test",
          "meta": [
            {
              "id": "O_ALTER_DB_SPEC",
              "def": {
                "collation": "utf8_general_ci"
              }
            }
          ]
        }
      }
    }
  ]
}
`;

exports[`Parser: Should alter test database with collation utf8_general_ci 4`] = `
{
  "id": "MAIN",
  "def": [
    {
      "id": "P_DDS",
      "def": {
        "id": "P_ALTER_DB",
        "def": {
          "database": "test",
          "meta": [
            {
              "id": "O_ALTER_DB_SPEC",
              "def": {
                "collation": "utf8_general_ci"
              }
            }
          ]
        }
      }
    }
  ]
}
`;

exports[`Parser: Should create test database even when having two collate options. 1`] = `
{
  "id": "MAIN",
  "def": [
    {
      "id": "P_DDS",
      "def": {
        "id": "P_ALTER_DB",
        "def": {
          "database": "test",
          "meta": [
            {
              "id": "O_ALTER_DB_SPEC",
              "def": {
                "collation": "utf8_cirylic_ci"
              }
            },
            {
              "id": "O_ALTER_DB_SPEC",
              "def": {
                "charset": "utf8"
              }
            },
            {
              "id": "O_ALTER_DB_SPEC",
              "def": {
                "collation": "utf8_general_ci"
              }
            }
          ]
        }
      }
    }
  ]
}
`;

exports[`Parser: Should create test database with collation utf8_general_ci and charset utf8 1`] = `
{
  "id": "MAIN",
  "def": [
    {
      "id": "P_DDS",
      "def": {
        "id": "P_ALTER_DB",
        "def": {
          "database": "test",
          "meta": [
            {
              "id": "O_ALTER_DB_SPEC",
              "def": {
                "collation": "utf8_general_ci"
              }
            },
            {
              "id": "O_ALTER_DB_SPEC",
              "def": {
                "charset": "utf8"
              }
            }
          ]
        }
      }
    }
  ]
}
`;
